### PR TITLE
feat(i18n): add peristant language preference sync for authenticated users

### PR DESCRIPTION
**Pull request by karthikjeeyar**
## Description

Implements bidirectional synchronization of language preferences between the app's language API and user storage, ensuring language settings persist across browser sessions for authenticated users.

### Backend
- Added `@backstage/plugin-user-settings-backend` dependency to enable persistent user settings storage
- Integrated user settings backend plugin with conditional loading based on persistence configuration

### Frontend  
- Created `useLanguagePreference` hook with comprehensive bidirectional sync logic
- Automatically detect browser's language settings to use it as a default lanaguage.
- Added extensive test coverage covering all edge cases
- Integrated hook into Root component to enable automatic language preference persistence
- Added configuration validation and database persistence checks

### Documentation
- Added [language preferences](https://github.com/redhat-developer/rhdh/blob/bdd4623f7e0193b23724db203d395cb3a2762200/docs/customization.md#language-preferences) section to customization documentation

## Key Features

- **Persistent Language Settings**: Language preferences automatically persist using database storage with no configuration required.
- **Default Language selection**: user's browser's langugage settings or admin configured value.
- **Optional Configuration**: Users can opt-out to browser storage if desired
- **Guest User Safe**: No sync operations for guest users 
- **Smart Sync Logic**: Prevents infinite loops and handles hydration gracefully
- **Error Resilient**: Graceful error handling with proper cleanup

## Configuration
Language persistence works out of the box with database storage. To opt-out to browser storage:
```
userSettings:
  persistence: browser # opt-out of database storage
```

## Notes
Language preferences now persist across browser sessions for authenticated users by default, providing a more consistent and personalized experience without requiring any configuration.

## Default language selection follows this priority order:

1. **Browser Language Priority**: The system first checks the user's browser language preferences to provide a personalized experience.

2. **Configuration Priority**: If no browser language matches the supported locales, the `defaultLocale` from the `i18n` configuration is used as a fallback.

3. **Fallback Priority**: If neither browser preferences nor configuration provide a match, defaults to `en`.

image

## Which issue(s) does this PR fix

- Fixes https://issues.redhat.com/browse/RHIDP-8658

## How to test

1. add the `i18n` config
  ```yaml
  i18n:
    locales:
      - en
      - de
      - fr
 ```
2. Sign in as an authenticated user (not guest)
3. Switch language using the language selector in the setting page.
4. Open in new tab/window in same browser - language preference should be maintained
5. Refresh the page on your different device/browser to see the preferred language
6. Test with guest user - language changes should not persist (expected behavior)

## PR acceptance criteria

Please make sure that the following steps are complete:

- [ ] GitHub Actions are completed and successful
- [x] Unit Tests are updated and passing
- [ ] E2E Tests are updated and passing
- [x] Documentation is updated if necessary (requirement for new features)
- [x] Add a screenshot if the change is UX/UI related

### CODE REVIEW

File: packages/backend/src/index.ts [hunk 1]
```diff
@@ -161,5 +161,6 @@ if (process.env.ENABLE_AUTH_PROVIDER_MODULE_OVERRIDE !== 'true') {
 backend.add(import('@internal/plugin-dynamic-plugins-info-backend'));
 backend.add(import('@internal/plugin-scalprum-backend'));
 backend.add(import('@internal/plugin-licensed-users-info-backend'));
+backend.add(import('@backstage/plugin-user-settings-backend'));
```
Comment (christoph-jerolimov) [inline packages/backend/src/index.ts]: Hi @karthikjeeyar, none of these (core) plugins has an opt-out at the moment. But since we're adding another one here, that could break a plugin or customer maybe in the future...

Can we have this enabled by default, but with an app-config opt-out option?

```yaml
user-settings:
  persistence: browser
```

Which means the backend isn't loaded? Not sure if your change in `app/src/apis.ts` must know and handle this as well?

Wdyt?

**Review by christoph-jerolimov — COMMENTED** [review]
No issue, just an idea / question:

File: packages/backend/src/index.ts (hunk 1 above)
Comment (karthikjeeyar) [reply packages/backend/src/index.ts]: That is a great suggestion!  I am now conditionally loading the backend plugin based on the app configuration and `app/src/apis.ts` is also updated to use the configuration to opt-out to browser storage. I have added documentation and added more details in the PR description.

File: packages/app/config.d.ts [hunk 2]
```diff
@@ -261,4 +261,16 @@ export interface Config {
     card: { [key: string]: string };
     full?: boolean;
   };
+
+  /**
+   * Allows you to customize user settings.
```
Comment (debsmita1) [inline packages/app/config.d.ts]:
```suggestion
   * Configuration options for your user settings.
```

**Comment by karthikjeeyar** [comment]
> Should we instead write the current language setting to the database as soon as the app loads with database persistence enabled?

I am now detecting user prefferend language  from browser's language settings and using it as the default language. 

Priority of this default language is as follows.

1.  **User centric**: navigator.languages (first available language set in the browser will be picked up)

2. **Admin config as a first fallback**: `defaultLocale` from app-config

3. if nothing found, default to english as final fallback.

**Comment by debsmita1** [comment]
Tested the following locally,
when the app loads for the first time, it auto-detects "English(India)" from my browser settings and loads it in the language dropdown. When I change it to some other language, it persists.

/lgtm

**Comment by karthikjeeyar** [comment]
Rebased to main branch and pushed again.

**Comment by HusneShabbir** [comment]
/lgtm
Tested locally, works as expected.
